cmd/kube-controller-manager/app/options/options.go

```diff
@@ -181,6 +181,8 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.Float32Var(&s.SecondaryNodeEvictionRate, "secondary-node-eviction-rate", 0.01, "Number of nodes per second on which pods are deleted in case of node failure when a zone is unhealthy (see --unhealthy-zone-threshold for definition of healthy/unhealthy). Zone refers to entire cluster in non-multizone clusters. This value is implicitly overridden to 0 if the cluster size is smaller than --large-cluster-size-threshold.")
 	fs.Int32Var(&s.LargeClusterSizeThreshold, "large-cluster-size-threshold", 50, "Number of nodes from which NodeController treats the cluster as large for the eviction logic purposes. --secondary-node-eviction-rate is implicitly overridden to 0 for clusters this size or smaller.")
 	fs.Float32Var(&s.UnhealthyZoneThreshold, "unhealthy-zone-threshold", 0.55, "Fraction of Nodes in a zone which needs to be not Ready (minimum 3) for zone to be treated as unhealthy. ")
+	fs.BoolVar(&s.EnableBootstrapSigner, "enable-bootstrap-signer", false, "Enables a controller that will sign kube-system/cluster-info configmap with bootstrap token.")
```

Comment:
@deads2k Anyway, that requires a refactor of all the controllers that already are in the controller-manager.
I strongly think we should do this, but in a separate PR in any case.
Refactoring the controller-manager, adding a common flag for which controllers should be enabled is fine, but will in either case cause deprecations of the existing cm enable-* flags. So if @jbeda promises to implement what federation has with the --controller option in the future, I think we should be safe.

Anyway, I think we can promise that we are just fine with removing these two flags as soon as the generic controller flag is in.

Of course, it's @jbeda's choice whether he wants to do it now or later this cycle